docs/src/references/c8y-configuration-management.md

```diff
@@ -162,4 +165,162 @@ Note that:
+      * `/http-root/config/desired/<child-id>/<config filename>`<br/>
+        for config-files to be consumed by the external device (desired state)
+  * `tedge_agent` notifies over MQTT any change of a file under `/http-root/` (e.g. when uploaded by a external-device or updated by a local process). The notification messages are published on the topic `tedge/filetransfer_change/{path}`.<br/>
+    TO-BE-DEFINED: Content of payload?
+      
+      Example:
+      * HTTP PUT URL: `http://<thin-edge IP address>/tedge/filetransfer/config/current/child1/bar.conf`
+      * relating MQTT notification topic: `/tedge/filetransfer_change/config/current/child1/bar.conf`
+
+  * `c8y_configuration_plugin` allows to handle two versions for each configuration file (_current_ and _desired_).
+    * The configuration file's record `path` is treated as path of the _current_ version of a configuration file (see to section 'Configuration' above).
+    * The configuration file's record `desired` can be used to define the path of the _desired_ version of a configuration file. If `desired` is not defined, then `path` is treated as _desired_ version also.
+    * On c8y read request for a configuration, `c8y_configuration_plugin` returns the `current` file content.
+    * On c8y write request for a configuration, `c8y_configuration_plugin` updates the `desired` file content and waits for the `current` content to be updated to return the operation status to cumulocity.  
+    * The current behavior of c8y_configuration_plugin can be simulated with desired = current = path when only a path is provided.
+
+Example Plugin Config:
+```shell
+$ cat /etc/tedge/c8y/c8y-configuration-plugin.toml
+files = [
+    { type = 'bar.conf', child_id = 'child1', current = '/http-root/config/current/child1/bar.conf', desired = '/http-root/config/desired/child1/bar.conf' } }
+  ]
+```
+
+Example Flow:
+
+**Start Behaviour:**
+  * external device child1: starts
+  * external device child1: sends all its config files with HTTP PUT requests to the thin-edge device; 
+
+    Example: HTTP PUT to `http://<thin-edge IP address>/tedge/filetransfer/current/child1/bar.conf`
+
+**Device-to-Cloud Behaviour:**
+  * at some point a config retrieval for type `bar.conf` for `child1` arrives at C8Y config plugin
+  * C8Y config plugin: sends the file `/http-root/config/current/child1/bar.conf` to C8Y<br/><br/>
+    NOTE: The responsibility to assure that the latest config file content on the thin-edge device is always on the external device. When there is no file the plugin sends an error to the cloud on an every incoming config retrieval request.
+
+**Cloud-to-Device Behaviour:**
+  * at some point a config sent from cloud for type `bar.conf` for `child1` arrives at C8Y config plugin
+  * C8Y config plugin: stores the files received from C8Y to `/http-root/config/desired/child1/bar.conf`
```

Comment:
In this sentence, the file will be stored in the tedge device, right? Why not `/tmp`, it is already configurable by our tedge CLI, and this is what we do for parent device config download.

Reply:
Applied. Now `/tmp` is used.